submissions/YuliiaHordiichuk/post-oop/index.js

```diff
@@ -0,0 +1,80 @@
+/* Refer to https://github.com/OleksiyRudenko/a-tiny-JS-world for the task details
+   Complete the below for code reviewers' convenience:
+
+   Code repository: https://github.com/YuliiaHordiichuk/a-tiny-JS-world/tree/post-oop
+   Web app: https://yuliiahordiichuk.github.io/a-tiny-JS-world/
+   */
+
+// ======== OBJECTS DEFINITIONS ========
+// Define your objects here
+
+class Inhabitant {
+   constructor(name, gender, saying, species) {
+      this.name = name; 
+      this.gender = gender; 
+      this.saying = saying; 
+      this.species = species; 
+   };
+
+   getMessage() {
+      const messageText = [
+      this.saying,
+      `I am a ${this.species}.`, 
+      `My name is ${this.name}.`,
+      `Nice to meet you!`,
+      `My gender is ${this.gender}.`
+   ]; 
+
+   return messageText.join(' ');  
+   }
+}
+
+class Human extends Inhabitant {
+   constructor(name, gender, saying, species) {
+      super(name, gender, saying, species); 
+      this.hands = 2; 
+      this.legs = 2; 
+   }
+
+   getMessage () {
+      return `${super.getMessage()} I have ${this.legs} legs and ${this.hands} hands.` 
+   }
+}
+
+class Animal extends Inhabitant {
+   constructor(name, gender, saying, species) {
+      super(name, gender, saying, species); 
+      this.legs = 4; 
+   }
+
+   getMessage () {
+      return `${super.getMessage()} I have ${this.legs} legs.`
+   }
```

Comment:
All inhabitants have legs. Why handling them needs a different approach compared with name, gender, species and saying?

Reply:
You are right, done